### ISSUE TITLE
Wrong normalization of PolyRatFun

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#38]
The bug introduced in a795a9c315dc75d1f5ca0f380becba8e2259d774 is still there.
```
CF num,rat;
S n1,x;
L F = num(n1)*num(1/2);
id num(x?) = rat(x,1);
P "%t";
ModuleOption PolyRatFun=rat;
P +s;
.end
```
somehow gives
```
 + rat(n1,1)
   F =
       + rat(n1,1)
      ;
```
(I thought it was fixed in 4f6cb5513c4e508f87e1e134b711f28f839e9c64, but not.)

**Comment by vermaseren** [comment]
It looks like poly_ratfun_normalize does not take fractions from the numerator.
I will sneak in some code somewhere, borrowing from MakeInteger.
May take a day.

Jos

**Comment by tueda** [comment]
The above example works after ffabb372610f3cea119b0bf7cfeea7bc27db3762, but there is still a sign problem. If I put a minus sign:
```
CF num,rat;
S n1,x;
L F = num(n1)*num(-1/2);
id num(x?) = rat(x,1);
P "%t";
ModuleOption PolyRatFun=rat;
P +s;
.end
```
the latest version gives
```
 + rat(n1,2)
   F =
       + rat(n1,2)
      ;
```
So the sign is ignored.

**Comment by tueda** [comment]
Fixed in 1470321b7961447e9094bbf853e3a3c2f7448367.

**Comment by tueda** [comment]
Now another problem probably related to `poly_ratfun_normalize()`:
```
CF rat;
PolyRatFun rat;
L F = rat(1,1) - rat(1,1);
P +s;
.end
```
gives
```
   F =
       + rat(1,1)
       + rat(1,1)*rat( - 1,1)
      ;
```
So the rational functions are not combined.

**Comment by tueda** [comment]
The above one was fixed in 4cec87deea15e90e465266ee5521b261bdbd5233. The next one that gives a wrong result is:
```
CF rat;
S ep,x;
PolyRatFun rat;
L F = rat(x,1)*rat(1+ep,1);
P +s;
.end
```
which gives
```
   F =
       + rat(ep*x,1)
      ;
```

And less importantly, FORM now doesn't complain dot products in PolyRatFun, but just freezes:
```
V a;
CF rat;
PolyRatFun rat;
L F = rat(a.a,1);
P +s;
.end
```

**Comment by tueda** [comment]
The multiplication of `rat` functions was fixed in 1fd92ca0166772498a063c12ed29bd1636c11a83 and so I close this issue. For the problem of the freeze with dot products in PolyRatFun, I will make a separate issue #39.
